src/code/z_demo.c

```diff
@@ -1,99 +1,1634 @@
+    func_800EA258,      // CS_STATE_1
+    Cutscene_DoNothing, // CS_STATE_2
+    func_800ED9C4,      // CS_STATE_3
+    Cutscene_DoNothing, // CS_STATE_4
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAC44.s")
+void Cutscene_Update1(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    if (gSaveContext.cutscene < 0xFFF0) {
+        sCsStateHandlers1[csCtx->state](globalCtx, csCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAC94.s")
+CutsceneStateHandler sCsStateHandlers2[] = {
+    Cutscene_DoNothing, // CS_STATE_0
+    func_800EA2B8,      // CS_STATE_1
+    func_800ED980,      // CS_STATE_2
+    func_800EDA04,      // CS_STATE_3
+    func_800ED980,      // CS_STATE_4
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAD14.s")
+void Cutscene_Update2(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    if ((gSaveContext.cutsceneTrigger != 0) && (globalCtx->sceneLoadFlag == 0x14)) {
+        gSaveContext.cutsceneTrigger = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAD48.s")
+    if ((gSaveContext.cutsceneTrigger != 0) && (csCtx->state == CS_STATE_0)) {
+        gSaveContext.cutscene = 0xFFFD;
+        gSaveContext.cutsceneTrigger = 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAD7C.s")
+    if (gSaveContext.cutscene >= 0xFFF0) {
+        func_800EDA84(globalCtx, csCtx);
+        sCsStateHandlers2[csCtx->state](globalCtx, csCtx);
+    }
+}
 
+void Cutscene_DoNothing(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+}
+
+s32 func_800EA220(GlobalContext* globalCtx, CutsceneContext* csCtx, f32 target) {
+    return Math_StepToF(&csCtx->unk_0C, target, 0.1f);
+}
+
+void func_800EA258(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    Interface_ChangeAlpha(1);
+    ShrinkWindow_SetLetterboxTarget(0x20);
+    if (func_800EA220(globalCtx, csCtx, 1.0f)) {
+        Audio_SetCutsceneFlag(true);
+        csCtx->state++;
+    }
+}
+
+void func_800EA2B8(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    func_800ED980(globalCtx, csCtx);
+    Interface_ChangeAlpha(1);
+    ShrinkWindow_SetLetterboxTarget(0x20);
+    if (func_800EA220(globalCtx, csCtx, 1.0f)) {
+        Audio_SetCutsceneFlag(true);
+        csCtx->state++;
+    }
+}
+
+void Cutscene_Command_Misc(GlobalContext* globalCtx2, CutsceneContext* csCtx, CsCmdBase* cmd) {
+    static u16 D_801BB15C = 0xFFFF;
+    Player* player = GET_PLAYER(globalCtx2);
+    GlobalContext* globalCtx = globalCtx2;
+    u8 isStartFrame = false;
+    f32 progress;
+    SceneTableEntry* loadedScene;
+    u16 time;
+
+    if ((csCtx->frames < cmd->startFrame) || (csCtx->frames >= cmd->endFrame && cmd->endFrame != cmd->startFrame)) {
+        return;
+    }
+
+    progress = Environment_LerpWeight(cmd->endFrame - 1, cmd->startFrame, csCtx->frames);
+    if (csCtx->frames == cmd->startFrame) {
+        isStartFrame = true;
+    }
+
+    switch (cmd->base) {
```

Comment:
No. I think I didn't add one because this function didn't match when I originally made the PR.
I think I could make one, but most of the values of that enum would be placeholders, because I don't know what many of those cases do.